src/main.rs

```diff
@@ -145,14 +148,23 @@ fn main() -> Result<()> {
     Ok(())
 }
 
-fn disable_bpf_stats_procfs() -> Result<()> {
+fn procs_bfs_stats_disable() -> Result<()> {
     fs::write(PROCFS_BPF_STATS_ENABLED, b"0").context(format!(
         "Failed to disable BPF stats via {}",
         PROCFS_BPF_STATS_ENABLED
     ))?;
     Ok(())
 }
 
+fn procfs_bpf_stats_is_enabled() -> Result<bool> {
+    let value = fs::read_to_string(PROCFS_BPF_STATS_ENABLED)
+        .context(format!("Failed to read from {}", PROCFS_BPF_STATS_ENABLED))?
+        .trim()
+        .to_string();
+
+    Ok(value == "1")
+}
```

Comment:
Avoid (re-)allocating by using the buffer `read_to_string` gives you
```suggestion
fn procfs_bpf_stats_is_enabled() -> Result<bool> {
    fs::read_to_string(PROCFS_BPF_STATS_ENABLED)
        .context(format!("Failed to read from {}", PROCFS_BPF_STATS_ENABLED))
        .map(|value| value.trim() == "1")
}
```

This is because `trim` returns a slice of the `String` `read_to_string` gives you, so instead of allocating with `to_string`, just compare the slices instead.